pkg/config/autodiscovery/autodiscovery.go

```diff
@@ -81,19 +81,15 @@ func DiscoverComponentsFromEnv() ([]config.ConfigurationProviders, []config.List
 		return detectedProviders, detectedListeners
 	}
 
-	if config.IsFeaturePresent(config.Docker) || config.IsFeaturePresent(config.Containerd) || config.IsFeaturePresent(config.Podman) || config.IsFeaturePresent(config.ECSFargate) {
-		detectedProviders = append(detectedProviders, config.ConfigurationProviders{Name: names.Container, Polling: true, PollInterval: "1s"})
-		if !config.IsFeaturePresent(config.Kubernetes) {
-			detectedListeners = append(detectedListeners, config.Listeners{Name: names.Container})
-			log.Info("Adding Container listener from environment")
-		}
-		log.Info("Adding Container provider from environment")
-	}
-
+	// Either add kubelet listener/provider or container listener/provider, not both
```

Comment:
At first sight, I believe this change will break some cases in AD. AD has always worked with multiple providers, the only requirement is to have a single listener, hence the code that was there before.
Typically, check configurations that are defined in Dockerfile labels are going to be broken by this change (when running inside Kubernetes).

Also, I don't expect the providers to provide duplicated configs, as the `container` provider looks at container labels and the `kubelet` provider looks at POD annotations, but maybe I miss a case?

Finally, if there's for some reason a hard requirement on having a single provider, I believe the proper way is to include the work done by the `kubelet` provider in the `container` listener.

Reply:
> Finally, if there's for some reason a hard requirement on having a single provider, I believe the proper way is to include the work done by the kubelet provider in the container listener.

I thought that this PR would be an easier change, but you're correct. Fortunately it seems to be an easy change. I see #12851, so let's move the discussion there.